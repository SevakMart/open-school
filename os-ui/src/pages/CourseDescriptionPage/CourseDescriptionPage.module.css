.mainContent {
  margin: 48px 88px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}/*# sourceMappingURL=CourseDescriptionPage.module.css.map */