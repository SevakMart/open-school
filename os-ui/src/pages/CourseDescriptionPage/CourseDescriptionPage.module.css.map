{"version":3,"sources":["CourseDescriptionPage.module.scss","../../styles/Mixins/FlexMixins.scss","CourseDescriptionPage.module.css"],"names":[],"mappings":"AAEA;EACI,iBAAA;ECFF,aAAA;EACA,mBDEqB;ECDrB,mBDCyB;ECAzB,8BAAA;ACEF","file":"CourseDescriptionPage.module.css"}