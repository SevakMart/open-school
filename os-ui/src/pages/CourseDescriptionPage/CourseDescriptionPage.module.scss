@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";

.navAndMainTitle {
    height:15rem;
    h1 {
        padding-left:6%;
        position:absolute;
        top:7rem;
    }
}

.mainContent {
    padding-left:6%;
    padding-right:6%;
    @include flexStyle(row,none, space-between);
}