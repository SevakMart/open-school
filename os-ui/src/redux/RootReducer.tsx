import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
