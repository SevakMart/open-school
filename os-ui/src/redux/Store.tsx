import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
