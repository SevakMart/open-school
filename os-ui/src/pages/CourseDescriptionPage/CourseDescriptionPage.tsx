import { useEffect, useState, useMemo } from 'react';
import { useLocation, useParams, useNavigate } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';

import { RootState } from '../../redux/Store';
import courseService from '../../services/courseService';
import userService from '../../services/userService';
import NavbarOnSignIn from '../../component/Navbar-Component/NavbarOnSignIn/NavbarOnSignIn';
import CourseMainContent from './Subcomponents/CourseMainContent/CourseMainContent';
import CourseSummary from './Subcomponents/CourseSummary/CourseSummary';
import Modal from '../../component/Modal/Modal';
import ModalMessageComponent from './Subcomponents/CourseSummary/Subcomponent/ModalMessageComponent/ModalMessageComponent';
import { CourseDescriptionType } from '../../types/CourseDescriptionType';
import styles from './CourseDescriptionPage.module.scss';

const CourseDescriptionPage = () => {
  const userInfo = useSelector<RootState>((state) => state.userInfo);
  const [courseInfo, setCourseInfo] = useState({});
  const [errorMessage, setErrorMessage] = useState('');
  const [enrollmentErrorMessage, setEnrollmentErrorMessage] = useState('');
  const [showEnrollmentMessage, setShowEnrollmentMessage] = useState(false);
  const [isEnrolled, setIsEnrolled] = useState(false);
  const location = useLocation();
  const navigate = useNavigate();
  const { t } = useTranslation();
  const { courseId } = useParams();
  const params = new URLSearchParams(location.search);
  const idAndToken = useMemo(() => ({
    token: (userInfo as any).token,
    id: (userInfo as any).id,
  }), []);
  const {
    title, description, goal, modules, mentorDto, rating, enrolled, level, language, duration,
  } = courseInfo as CourseDescriptionType;
  const { mainContent, navAndMainTitle } = styles;

  const enrollInCourse = () => {
    userService.enrollCourse(idAndToken.id, Number(courseId), idAndToken.token)
      .then(() => setShowEnrollmentMessage(true))
      .catch(() => setEnrollmentErrorMessage(t('messages.error')));
  };

  const removeModalMessageAfterCourseEnrollment = () => {
    params.set('enrolled', 'true');
    setIsEnrolled(true);
    setShowEnrollmentMessage(false);
    navigate(`${location.pathname}?${params}`);
  };

  useEffect(() => {
    if (params.has('enrolled')) setIsEnrolled(true);
    courseService.requestCourseDescription(Number(courseId), {}, idAndToken.token)
      .then((courseDescriptionData) => setCourseInfo({ ...courseDescriptionData }))
      .catch(() => setErrorMessage(t('messages.error')));
  }, []);
  return (
    <>
      <div className={navAndMainTitle}>
        <NavbarOnSignIn />
        <h1>{title}</h1>
      </div>
      {!errorMessage && Object.keys(courseInfo).length ? (
        <div className={mainContent}>
          <CourseMainContent
            description={description}
            goal={goal}
            mentorDto={mentorDto}
            modules={modules}
          />
          <CourseSummary
            rating={rating}
            enrolled={enrolled}
            level={level}
            language={language}
            duration={duration}
            enrollInCourse={enrollInCourse}
            isEnrolled={isEnrolled}
            courseId={Number(courseId)}
            userIdAndToken={idAndToken}
          />
        </div>
      ) : <h2>{errorMessage}</h2>}
      {
        !enrollmentErrorMessage && showEnrollmentMessage
          ? (
            <Modal>
              <ModalMessageComponent
                enrollInCourse={removeModalMessageAfterCourseEnrollment}
              />
            </Modal>
          )
          : <h2>{enrollmentErrorMessage}</h2>
      }
    </>
  );
};
export default CourseDescriptionPage;
