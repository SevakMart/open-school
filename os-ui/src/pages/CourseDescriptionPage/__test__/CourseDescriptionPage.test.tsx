import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../redux/Store';
import CourseDescriptionPage from '../CourseDescriptionPage';

describe('Create test cases for course description page', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(<Provider store={store}>
      <CourseDescriptionPage />
      {/* eslint-disable-next-line */}
      </Provider>,{ wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});
