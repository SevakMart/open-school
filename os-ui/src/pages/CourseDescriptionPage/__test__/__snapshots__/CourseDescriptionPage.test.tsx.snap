// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for course description page Create a snapshot test 1`] = `
<DocumentFragment>
  <nav
    class="mainContent"
  >
    <h2>
      string.homePage.navBar.logo
    </h2>
    <div
      class="navMainContent"
    >
      <p>
        string.homePage.navBar.allLearningPaths
      </p>
      <p>
        string.homePage.navBar.myLearningPaths
      </p>
      <p>
        string.homePage.navBar.mentors
      </p>
      <div
        class="bookmarkIconStyle"
      >
        <img
          alt="bookmark"
          src="bookMark.svg"
          style="font-size: 20px; cursor: pointer; fill: none;"
        />
      </div>
      <img
        alt="Notification"
        class="notificationIcon"
        src="Notification.svg"
      />
      <div
        class="userInfoContent"
      >
        <img
          alt="avatar"
          class="avatarLogo"
          src="https://reactjs.org/logo-og.png"
        />
        <img
          alt="Arrow down Icon"
          class="arrowDownIcon"
          src="Arrow-down.svg"
        />
      </div>
      <div />
    </div>
  </nav>
  <div
    class="mainContent"
  >
    <div
      class="loader"
    >
      <span />
      <span />
      <span />
    </div>
  </div>
</DocumentFragment>
`;
